x-pack/plugins/infra/public/pages/metrics/hosts/components/controls_content.tsx

```diff
@@ -44,21 +44,24 @@ export const ControlsContent: React.FC<Props> = ({
     if (!controlGroup) {
       return;
     }
-    if (
-      !isEqual(controlGroup.getInput().timeRange, timeRange) ||
-      !compareFilters(controlGroup.getInput().filters ?? [], filters) ||
-      !isEqual(controlGroup.getInput().query, query)
-    ) {
-      controlGroup.updateInput({
-        timeRange,
-        query,
-        filters,
-      });
-    }
-  }, [query, filters, controlGroup, timeRange]);
+    const filtersSubscription = controlGroup.onFiltersPublished$.subscribe((newFilters) => {
```

Comment:
@jennypavlova I moved the subscription creation into useEffect so that they can be canceled.

Reply:
Nice, thanks!